### PR TITLE
Updated Agreement List table with Next Budget Line

### PR DESCRIPTION
**Pull request by stevtek**
## What changed

* new/updated Agreements list columns with Next Budget Line 
* logic to find Next Budget Line using only PLANNED and IN_EXECUTION status

## Issue

#1655 #965 

## How to test

View the Agreements List to see the new Next Budget Line and update Next Need By columns

## Screenshots

image

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [ ] ~Form validations updated~

## Links

_If relevant, e.g. for a link to a piece of markdown documentation_

### CODE REVIEW

File: frontend/cypress/e2e/agreementList.cy.js [hunk 1]
```diff
@@ -6,21 +6,62 @@ beforeEach(() => {
...
 });
 
-afterEach(() => {
-    cy.injectAxe();
-    cy.checkA11y(null, null, terminalLog);
-});
+// afterEach(() => {
+//     cy.injectAxe();
+//     cy.checkA11y(null, null, terminalLog);
+// });
```
Comment (tdonaworth) [inline frontend/cypress/e2e/agreementList.cy.js:12]: Is this for testing, or not needed?

**Review by tdonaworth — APPROVED** [review]

**Review by fpigeonjr — APPROVED** [review]
Looks good Steve. Wonder if we need to include the fees in the Agreements list?
image

File: frontend/cypress/e2e/agreementList.cy.js (hunk 1 above)
Comment (stevtek) [reply frontend/cypress/e2e/agreementList.cy.js:12]: good catch, I was troubleshooting and forgot to revert

**Comment by stevtek** [comment]
> Wonder if we need to include the fees in the Agreements list?
It's not obvious in Figma, it's all round numbers

**Pull request merged**